lib/matplotlib/axes/_axes.py

```diff
@@ -599,6 +600,80 @@ def indicate_inset_zoom(self, inset_ax, **kwargs):
 
         return rectpatch, connects
 
+    @docstring.dedent_interpd
+    def secondary_xaxis(self, location, *, functions=None, **kwargs):
+        """
+        Add a second x-axis to this axes.
+
+        For example if we want to have a second scale for the data plotted on
+        the xaxis.
+
+        %(_secax_docstring)s
+
+        Examples
+        --------
+
+        Add a secondary axes that shows both wavelength for the main
+        axes that shows wavenumber.
+
+        .. plot::
+
+            fig, ax = plt.subplots()
+            ax.loglog(range(1, 360, 5), range(1, 360, 5))
+            ax.set_xlabel('wavenumber [cpkm]')
+
+
+            def invert(x):
+                return 1 / x
+
+            secax = ax.secondary_xaxis('top', functions=(invert, invert))
+            secax.set_xlabel('wavelength [km]')
+            plt.show()
+
+
+        """
+        if (location in ['top', 'bottom'] or isinstance(location, Number)):
+            secondary_ax = SecondaryAxis(self, 'x', location,
+                                          functions, **kwargs)
+            self.add_child_axes(secondary_ax)
+            return secondary_ax
+        else:
+            raise ValueError('secondary_xaxis location must be either '
```

Comment:
I think you can just let SecondaryAxis.set_location emit the proper error message?

Reply:
... could, and then the user would get an error message for something they didn't directly call...